src/pymmcore_widgets/_stage_widget.py

```diff
@@ -75,153 +190,141 @@ class StageWidget(QWidget):
         self._levels = levels
         self._device = device
+        self._poll_timer_id: int | None = None
+
         self._dtype = self._mmc.getDeviceType(self._device)
-        assert self._dtype in STAGE_DEVICES, f"{self._dtype} not in {STAGE_DEVICES}"
+        if self._dtype not in {DeviceType.Stage, DeviceType.XYStage}:
+            raise ValueError("This widget only supports Stage and XYStage devices.")
 
-        self._create_widget(step)
-        self._connect_events()
-        self._set_as_default()
+        is_2axis = self._dtype is DeviceType.XYStage
+        Y = "Y" if is_2axis else self._device
 
-        self.destroyed.connect(self._disconnect)
+        # WIDGETS ------------------------------------------------
 
-    def step(self) -> float:
-        """Return the current step size."""
-        return self._step.value()  # type: ignore
+        self._move_btns = StageMovementButtons(self._levels, is_2axis)
+        self._step = self._move_btns.step_size
 
-    def setStep(self, step: float) -> None:
-        """Set the step size."""
-        self._step.setValue(step)
-
-    def _create_widget(self, step: float) -> None:
-        self._step = QDoubleSpinBox()
-        self._step.setValue(step)
-        self._step.setMaximum(99999)
-        self._step.valueChanged.connect(self._update_ttips)
-        self._step.clearFocus()
-        self._step.setAttribute(Qt.WidgetAttribute.WA_MacShowFocusRect, 0)
-        self._step.setButtonSymbols(QSpinBox.ButtonSymbols.NoButtons)
-        self._step.setAlignment(AlignCenter)
-
-        self._btns = QWidget()
-        self._btns.setLayout(QGridLayout())
-        self._btns.layout().setContentsMargins(0, 0, 0, 0)
-        self._btns.layout().setSpacing(0)
-        for glyph, (row, col, *_) in self.BTNS.items():
-            btn = QPushButton()
-            btn.setAutoRepeat(True)
-            btn.setFlat(True)
-            btn.setFixedSize(self.BTN_SIZE, self.BTN_SIZE)
-            btn.setFocusPolicy(Qt.FocusPolicy.NoFocus)
-            btn.setCursor(Qt.CursorShape.PointingHandCursor)
-            setTextIcon(btn, glyph)
-            btn.clicked.connect(self._on_click)
-            self._btns.layout().addWidget(btn, row, col, AlignCenter)
-
-        self._btns.layout().addWidget(self._step, 3, 3, AlignCenter)
-        self._set_visible_levels(self._levels)  # type: ignore
-        self._set_xy_visible()
-        self._update_ttips()
-
-        self._readout = QLabel()
-        self._readout.setAlignment(AlignCenter)
-        self._update_position_label()
-
-        self._poll_cb = QCheckBox("poll")
-        self._poll_cb.setMaximumWidth(50)
-        self._poll_timer = QTimer()
-        self._poll_timer.setInterval(500)
-        self._poll_timer.timeout.connect(self._update_position_label)
-        self._poll_cb.toggled.connect(self._toggle_poll_timer)
+        self._poslabel_x = QDoubleSpinBox()
+        self._poslabel_x.setButtonSymbols(QSpinBox.ButtonSymbols.NoButtons)
```

Comment:
Similar to my above note, Do you think we could show buttons, not because we want people to use them, but because we want people to know that they can edit the value and the stage will react? Or do you have a better idea for conveying this idea?

Reply:
same note as above :) dislike the spinbuttons, like the hover effects, think it's quickly learnable.  spin buttons as a last resort, would prefer a different option